subprojects/build-cache/src/integTest/groovy/org/gradle/caching/local/internal/DirectoryBuildCacheCleanupIntegrationTest.groovy

```diff
@@ -55,370 +54,157 @@ class DirectoryBuildCacheCleanupIntegrationTest extends AbstractIntegrationSpec
-                test
-            }
-            dependencies {
-                test "com.example:included:1.0"
-            }
-            assertBuildCacheOverTarget {
-                dependsOn cacheable, configurations.test
-                doFirst {
-                    println configurations.test.files
-                }
-            }
-        """
-        when:
-        runMultiple(MAX_CACHE_SIZE*2)
-        and:
-        then:
-        // build cache hasn't been cleaned yet
-        calculateCacheSize(listCacheFiles()) >= MAX_CACHE_SIZE
+        run()
         def lastCleanupCheck = gcFile().makeOlder().lastModified()
 
         when:
-        markCacheDirForCleanup()
-        and:
-        // During the build, the build cache should be over the target still (the composite didn't clean it up)
-        withBuildCache().succeeds("assertBuildCacheOverTarget")
+        run()
         then:
-        // build cache has been cleaned up now
-        calculateCacheSize(listCacheFiles()) <= MAX_CACHE_SIZE
-        gcFile().lastModified() >= lastCleanupCheck
-    }
+        assertCacheWasNotCleanedUpSince(lastCleanupCheck)
 
-    def "composite build with buildSrc do not try to clean build cache mid build"() {
-        file("included/build.gradle") << """
-            apply plugin: 'java'
-            group = "com.example"
-            version = "2.0"
-        """
-        // Copy cache configuration
-        file("included/buildSrc/settings.gradle").text = settingsFile.text
-        file("included/settings.gradle").text = settingsFile.text
-
-        settingsFile << """
-            includeBuild file("included/")
-        """
-        buildFile << """
-            configurations {
-                test
-            }
-            dependencies {
-                test "com.example:included:1.0"
-            }
-            
-            assertBuildCacheOverTarget {
-                dependsOn cacheable, configurations.test
-                doFirst {
-                    println configurations.test.files
-                }
-            }
-        """
         when:
-        runMultiple(MAX_CACHE_SIZE*2)
-        and:
+        lastCleanupCheck = markCacheForCleanup()
+        run()
         then:
-        // build cache hasn't been cleaned yet
-        calculateCacheSize(listCacheFiles()) >= MAX_CACHE_SIZE
-        def lastCleanupCheck = gcFile().makeOlder().lastModified()
-
-        when:
-        markCacheDirForCleanup()
-        and:
-        // During the build, the build cache should be over the target still (the composite didn't clean it up)
-        withBuildCache().succeeds("assertBuildCacheOverTarget", "-i")
-        then:
-        // build cache has been cleaned up now
-        calculateCacheSize(listCacheFiles()) <= MAX_CACHE_SIZE
-        gcFile().lastModified() >= lastCleanupCheck
+        assertCacheWasCleanedUpSince(lastCleanupCheck)
     }
 
+    @Ignore("Looks like GradleBuild doesn't work with build operations")
```

Comment:
I guess we need to fix this test in some way?

Reply:
I'll raise that issue in some other place.

Reply:
I've mentioned that it's related to https://github.com/gradle/gradle/issues/3983. I think it's okay to re-enable it once that's fixed. WDYT?

Reply:
Thats fine with me.